### PR TITLE
Prevent adding an empty cell

### PR DESCRIPTION
**Pull request by hbcarlos**
## References

Solves #15

## Code changes

Added a new argument in `IModelFactory` to prevent track if the notebook is initialized or not to prevent the `StaticNotebook` from adding a new empty cell.

### CODE REVIEW

File: packages/docregistry/src/default.ts [hunk 1]
```diff
@@ -227,12 +227,14 @@ export class TextModelFactory implements DocumentRegistry.CodeModelFactory {
    * Create a new model.
    *
    * @param languagePreference - An optional kernel language preference.
+   * @param modelDB - An optional modelDB.
```
Comment (jtpio) [inline packages/docregistry/src/default.ts:230]: We might as well add the optional `isInitialized` param to the docstring too?

Comment (hbcarlos) [reply packages/docregistry/src/default.ts:230]: I'm gonna wait for the tests to finish and I'll add a commit.

File: packages/docregistry/src/default.ts [hunk 2]
```diff
@@ -227,12 +227,15 @@ export class TextModelFactory implements DocumentRegistry.CodeModelFactory {
...
+   * @param isInitialized - An optional flag to check if the model is initialized.
    *
    * @returns A new document model.
    */
   createNew(
     languagePreference?: string,
-    modelDB?: IModelDB
+    modelDB?: IModelDB,
+    isInitialized?: boolean
```
Comment (jtpio) [inline packages/docregistry/src/default.ts:238]: Note to self: check whether this new parameter should be considered a breaking change.

Comment (echarles) [reply packages/docregistry/src/default.ts:238]: It is tagged as optional with `?`, so it should be backwards compatible.

**Review by echarles — APPROVED** [review]
LGTM. Discussed this with @jtpio and @hbcarlos and we gonna merge. If needed, we can revisit the method signature.

**Pull request merged**